### PR TITLE
[FIX] Timeseries: time_variable now created in Timeseries

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Some widgets (**Seasonal Adjustment**, **Moving Transform**, and **Difference**) create `Timeseries` when they get `Table` and they do not set `time_variable`.

##### Description of changes
Scripting part which creates `time_variable` is now moved from widget **As Timeseries** to `Timeseries`.
Now this code can be used when creating `Timeseries` from `Table`.

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/timeseries/timeseries.py [hunk 1]
```diff
@@ -24,7 +24,22 @@ def __init__(self, *args, **kwargs):
             self.time_variable = next(var for var in self.domain.attributes
                                       if isinstance(var, TimeVariable))
         except (StopIteration, AttributeError):
-            pass
+            try:
+                self.time_variable = next(var for var in self.domain.attributes
+                                          if isinstance(var, ContinuousVariable))
```
Comment (kernc) [inline orangecontrib/timeseries/timeseries.py]: I don't think any ContinuousVariable will cut it.

**Comment by kernc** [comment]
I don't like this. Where does it happen?

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3-timeseries/issues/277426278/

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3-timeseries/issues/420934579/

**Comment by kernc** [comment]
I don't like this change. `.time_variable` can be `None`. The failure happens on https://github.com/biolab/orange3-timeseries/pull/33/files#diff-baa207cfea1077bb261faf9efcf809c8R93. How about if the assertion `assert var in self.domain` was changed to:
```py
if var is None:
    self.attributes = self.attributes.copy()
    self.attributes.pop('time_variable')
    return

assert var in self.domain
```
Would that work?

**Comment by jerneju** [comment]
@kernc : And you also do not want to have a conversion from `Table` to `Timeseries` in a scripting part?

**Comment by kernc** [comment]
Why, doesn't `Timeseries.from_data_table(table)` do that?

**Comment by kernc** [comment]
This looks safer to me. Thanks.

**Pull request merged**